#! /bin/sh /usr/share/dpatch/dpatch-run
## 04_Composite.dpatch by  Jérémy Bobbio <jeremy.bobbio@etu.upmc.fr>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix an issue when runing inside a Composite-enabled X environment.

@DPATCH@
diff -urNad electricsheep-2.6.8~/mpeg2dec/libvo/video_out_x11.c electricsheep-2.6.8/mpeg2dec/libvo/video_out_x11.c
--- electricsheep-2.6.8~/mpeg2dec/libvo/video_out_x11.c	2006-06-26 21:28:08.000000000 +0200
+++ electricsheep-2.6.8/mpeg2dec/libvo/video_out_x11.c	2006-10-07 16:47:21.000000000 +0200
@@ -508,9 +508,13 @@
 
     /* find the visual with the highest depth */
     XvisualInfo = XvisualInfoTable;
-    for (i = 1; i < number; i++)
+    for (i = 1; i < number; i++) {
+        /* Skip ARGB visuals. */
+        if (XvisualInfoTable[i].depth > 24)
+	    continue;
 	if (XvisualInfoTable[i].depth > XvisualInfo->depth)
 	    XvisualInfo = XvisualInfoTable + i;
+    }
 
     instance->vinfo = *XvisualInfo;
     XFree (XvisualInfoTable);
